### PR TITLE
Add installer

### PR DESCRIPTION
**Pull request by theoturner**
Adds installer automating the build steps, including the slightly awkward `DOPENSSL_ROOT_DIR` on macOS. Readme updated accordingly.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    xcode-select --version || xcode-select --install
+    brew --version || yes | /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+    python3.7 --version || brew install python
+    cmake --version || brew install cmake
+    brew install openssl
+    for version in `ls -t /usr/local/Cellar/openssl/`; do
+        ossl=/usr/local/Cellar/openssl/$version
+        break
+    done
+fi
```
Comment (ejfitzgerald) [inline install.sh]: I think that it would be good to catch the `else` case here and exit with an error.

**Review by ejfitzgerald — CHANGES_REQUESTED** [review]
Hi Theo,

A really interesting change. In general I like this approach but I would recommend renaming the file to `quickstart.sh` or something similar. Can you also move the script into the `scripts/` folder.

File: install.sh (hunk 1 above)
Comment (theoturner) [reply install.sh]: Renamed and moved. Catch error if the openssl install is not present. Also added minor change if script run from scripts dir, jump up to root dir and continue as before.

**Comment by theoturner** [comment]
Final commit is merge of 3f8cdb8

**Comment by ejfitzgerald** [comment]
Sorry @theoturner we had a CI issue which is preventing this to be merged. Could you merge in the latest changes from `master` and then we can land this feature.

**Comment by theoturner** [comment]
@ejfitzgerald merged and changed `make -j` to `make -j 4` in the installer.

**Review by ejfitzgerald — APPROVED** [review]

**Comment by ruthbr** [comment]
I'd probably just change '(in the case where you have been running a local network)' to just '(if you have been running a local network)'

**Pull request merged**